File: solutions/2091-removing-minimum-and-maximum-from-array/array.js
// time: O(n)
// space: O(1)

/**
 * @param {number[]} nums
 * @return {number}
 */
const minimumDeletions = nums => {
    const n = nums.length;
    let min = 0;
    let max = 0;

    for (let i = 0; i < n; i++) {
        if (nums[min] > nums[i]) min = i;
        if (nums[max] < nums[i]) max = i;
    }

    if (min > max) [min, max] = [max, min];

    return Math.min(max + 1, min + 1 + n - max, n - min);
};

File: solutions/2091-removing-minimum-and-maximum-from-array/array.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
  public:
    int minimumDeletions(vector<int> nums) {
        const int n = nums.size();
        int mn = 0;
        int mx = 0;

        for (int i = 0; i < n; i++) {
            if (nums[i] < nums[mn]) mn = i;
            if (nums[i] > nums[mx]) mx = i;
        }

        if (mn > mx) {
            mx ^= mn;
            mn ^= mx;
            mx ^= mn;
        }

        return min(mn + 1 + n - mx, min(mx + 1, n - mn));
    }
};
